### PR TITLE
Closing dialog does not cancel (Manjaro Linux) (fix #154719)

### LINKED ISSUE #154719: Closing dialog does not cancel (Manjaro Linux)
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.68.1
- OS Version: Manjaro Linux 21.3.1

Steps to Reproduce:

1. Create a new text file (`Ctrl+N`)
2. Set the language mode to something you don't have a formatter for (e.g. LaTeX)
3. Invoke _Format document_ (`Ctrl+Shift+I`)
4. A dialog opens, which says “There is no formatter for latex files installed.” and allows the user to a) install one or b) cancel.
5. Close the modal either by pressing `Esc`, by pressing the **X** on the window decoration, or by pressing `Ctrl+F4`
6. The Extensions pane opens, even though the user wanted to cancel.

I can't reproduce this using [Codespaces / VNC](https://github.com/microsoft/vscode/blob/main/.devcontainer/README.md) on _main_, but on multiple machines running Manjaro.

Some relevant code snippets:
https://github.com/microsoft/vscode/blob/54dec24a4d1e67dc6cb7fae0e6119dc84e0fab92/src/vs/workbench/contrib/format/browser/formatActionsNone.ts#L68-L75
https://github.com/microsoft/vscode/blob/54dec24a4d1e67dc6cb7fae0e6119dc84e0fab92/src/vs/platform/dialogs/common/dialogs.ts#L223-L230

### PR DESCRIPTION
**Pull request by bpasero**
fix #154719

### CODE REVIEW

File: src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEdit.contribution.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ class BulkEditPreviewContribution {
 			const choice = await this._dialogService.show(
 				Severity.Info,
 				localize('overlap', "Another refactoring is being previewed."),
-				[localize('cancel', "Cancel"), localize('continue', "Continue")],
+				[localize('continue', "Continue"), localize('cancel', "Cancel")],
```
Comment (jrieken) [inline src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEdit.contribution.ts:125]: doesn't this break checking of `choice` further down

File: src/vs/workbench/contrib/format/browser/formatActionsNone.ts [hunk 2]
```diff
@@ -68,7 +68,7 @@ registerEditorAction(class FormatDocumentMultipleAction extends EditorAction {
 			const res = await dialogService.show(
 				Severity.Info,
 				message,
-				[nls.localize('cancel', "Cancel"), nls.localize('install.formatter', "Install Formatter...")]
+				[nls.localize('install.formatter', "Install Formatter..."), nls.localize('cancel', "Cancel")]
```
Comment (jrieken) [inline src/vs/workbench/contrib/format/browser/formatActionsNone.ts:71]: same - `choice === 1` assumes you want to format. This is now `0`, right?

Comment (bpasero) [reply src/vs/workbench/contrib/format/browser/formatActionsNone.ts]: yeah, better to set `cancelId`.

Tbh this is Electron API that we basically just have exposed 1 to 1 with the way Electron decided to deal with cancellation. From the docs:

```ts
/**
 * The index of the button to be used to cancel the dialog, via the `Esc` key. By
 * default this is assigned to the first button with "cancel" or "no" as the label.
 * If no such labeled buttons exist and this option is not set, `0` will be used as
 * the return value.
 */
cancelId?: number;
```

Esp. the part around checking for a "label" is bad when we think about the fact that our dialogs get translated to other languages. 

I can do a pass to check for other cases where we do not set `cancelId` and maybe make that even a required property.

Comment (bpasero) [reply src/vs/workbench/contrib/format/browser/formatActionsNone.ts]: via https://github.com/microsoft/vscode/pull/155049/commits/517600ee1f99f27abd202e62d69ecc600db20c68

File: src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEdit.contribution.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/contrib/bulkEdit/browser/preview/bulkEdit.contribution.ts:125]: via https://github.com/microsoft/vscode/pull/155049/commits/517600ee1f99f27abd202e62d69ecc600db20c68

**Review by jrieken — APPROVED** [review]

**Comment by jrieken** [comment]
Thanks

File: src/vs/workbench/contrib/format/browser/formatActionsNone.ts (hunk 2 above)
Comment (bpasero) [reply src/vs/workbench/contrib/format/browser/formatActionsNone.ts]: Yeah, similar to how we use `IAction` in notifications:

https://github.com/microsoft/vscode/blob/c9c6219f67ff76f10fe548bc1c5247ed3a91577a/src/vs/platform/notification/common/notification.ts#L127-L143

**Pull request merged**